[2025-11-07 Fiddler](https://thefiddler.substack.com/p/the-randy-hall-problem)
====================

Fiddler
-------
Assuming that there is a steady-state solution, let $p_2$ be the probability of being
at door 2, and $p_{13}$ be the probability of being at door 1 or door 3.  Let $p_r$,
what we want to solve for, be the probability of remaining at door 1 or door 3.

In [1]:
p_2, p_13, p_r = var("p_2,p_13,p_r")
s = solve([p_2 == 0.2*p_2 + (1-p_r)*p_13,
           p_13 == p_r*p_13 + 0.8*p_2,
           p_2 + p_13 == 1], [p_2, p_13])
show(s)
show(solve((p_2*2 == p_13).substitute(s), p_r))

[[p_2 == 5*(p_r - 1)/(5*p_r - 9), p_13 == -4/(5*p_r - 9)]]

[p_r == (3/5)]

So for $p_{13} = 2p_2$ in steady-state, $p_r = 0.6$.

Extra credit
------------
Assuming that there is a steady-state solution, let $p_2$ be the probability of being at door 2
after an even number of presses, and $p_{13}$ be the probability of being at door 1 or door 3
after an even number of presses.  Let $p_r$, what we want to solve for, be the probability of
remaining at door 1 or door 3.

In [2]:
s = solve([p_2 == p_2*(0.2*0.5 + 0.8*(1-p_r)) + p_13*(p_r*(1-p_r) + (1-p_r)*0.5),
           p_13 == p_2*(0.2*0.5 + 0.8*p_r) + p_13*(p_r*p_r + (1-p_r)*0.5),
           p_2 + p_13 == 1],
          [p_2, p_13])
show(s)
show(solve((p_2*2 == p_13).substitute(s), p_r))
assume(p_r >= 0, p_r <= 1)
show(solve((p_2*2 == p_13).substitute(s), p_r))

[[p_2 == 5*(2*p_r^2 - p_r - 1)/(10*p_r^2 - 13*p_r - 6), p_13 == -(8*p_r + 1)/(10*p_r^2 - 13*p_r - 6)]]

[p_r == -1/20*sqrt(181) + 1/20, p_r == 1/20*sqrt(181) + 1/20]

[p_r == 1/20*sqrt(181) + 1/20]

So for $p_{13} = 2p_2$ in steady-state, the positive solution is $p_r = \frac{1 + \sqrt{181}}{20}$,
approximately 72.3%.

Numerical simulations
---------------------
[Numerical simulations](20251107.go) agree.

    $ go run 20251107.go
    f:pr:0.500000 0.384439
    f:pr:0.600000 0.333608
    f:pr:0.700000 0.272446
    ec:pr:0.672681 0.376125
    ec:pr:0.722681 0.333431
    ec:pr:0.772681 0.287445
    ec:pr:0.672681 0.375617
    ec:pr:0.722681 0.333543
    ec:pr:0.772681 0.286921

Additional thought
------------------
What about after an odd number of presses?  In steady-state,

In [3]:
pr = (1+sqrt(181))/20
numerical_approx(1/3*0.2 + 2/3*(1-pr)), numerical_approx(1/3*0.8 + 2/3*pr)

(0.251545865097543, 0.748454134902457)

the probability of being at door 2 is only about 25.2%, significantly less
than after an even number of presses.

More thoughts
-------------
Considering the stability of the steady-state solutions.  Let the probability of
being at door 2 be $p_{2,n} = 1/3 + \delta_n$, so the probability of being at door 1 or
door 3 is $p_{13,n} = 2/3 - \delta_n$.  Then,
$\delta_{n+1} = 0.2\cdot\delta_n - (1-p_r)\cdot\delta_n = -0.2\cdot\delta_n$, so it
looks like the steady-state solution is stable and convergence should be pretty quick.

Similarly for the extra credit,
$\delta_{n+1} = (0.2\cdot0.5+0.8\cdot(1-p_r))\cdot\delta_n - (p_r(1-p_r) + 0.5\cdot(1-p_r))\delta_n \approx -0.2399\cdot\delta_n$,
which is also stable and should converge almost as quickly.  (Or more quickly, in terms of
the number of even _and_ odd presses.)

In [4]:
numerical_approx((0.2*0.5 + 0.8*(1-pr)) - (pr*(1-pr) + 0.5*pr))

-0.239898645178108